app/javascript/Search/Search.jsx

```diff
@@ -73,9 +73,13 @@ export class Search extends Component {
     return event.altKey || event.ctrlKey || event.metaKey || event.shiftKey;
   };
 
-  submit = (event) => {
-    if (hasInstantClick) {
-      event.preventDefault();
+  submit = () => {
+    const { searchTerm } = this.props;
+    this.enableSearchPageChecker = false;
+
+    if (hasInstantClick()) {
+      preloadSearchResults({ searchTerm });
+      displaySearchResults({ searchTerm });
     }
   };
```

Comment:
```suggestion
  submit = () => {
    event.preventDefault();
    
    const { value: searchTerm } = this.searchInputRef.current;
    const { searchTerm: currentSearchTerm } = this.props;
    
    this.enableSearchPageChecker = false;

    if (hasInstantClick() && searchTerm !== currentSearchTerm) {
      const { setSearchTerm } = this.props;
      setSearchTerm(searchTerm);
      
      preloadSearchResults({ searchTerm });
      displaySearchResults({ searchTerm });
    }
  };
```

refactor (blocking) - we need to tweak this a bit more to bring it inline with the current search submit behaviour, and remove flickering. If you check out the video below, you'll see there's some flickering of the search input currently while the 'normal' form submit happens and this function also runs. We can remove the flicker with these changes, to make it similar to the current `search` function


https://user-images.githubusercontent.com/20773163/126623344-788f1d8c-66c6-4fa6-ad72-97a50190177f.mp4